rust/src/lib/nm/nm_dbus/gen_conf/route_rule.rs

```diff
@@ -1,3 +1,47 @@
 // SPDX-License-Identifier: Apache-2.0
 
-// TODO
+use std::collections::HashMap;
+
+use super::super::NmIpRouteRule;
+
+const DEFAULT_IPV4: &str = "0.0.0.0";
+const DEFAULT_IPV6: &str = "::";
+const DEFAULT_IP_PREFIX: u8 = 0;
+const DEFAULT_ROUTE_TABLE: u32 = 254;
+const AF_INET: i32 = 2;
+
+impl NmIpRouteRule {
+    pub(crate) fn to_keyfile(&self) -> HashMap<String, String> {
+        let mut ret = HashMap::new();
+        if let Some(priority) = self.priority.as_ref() {
+            let prio_str = format!("priority {}", priority);
+            let from_str = match (self.from.as_ref(), self.from_len.as_ref()) {
+                (Some(f), Some(f_len)) => format!("from {}/{}", f, f_len),
+                (Some(f), None) => format!("from {}", f),
+                _ => format!(
+                    "from {}/{}",
+                    if self.family.unwrap_or_default() == AF_INET {
+                        DEFAULT_IPV4
+                    } else {
+                        DEFAULT_IPV6
+                    },
+                    DEFAULT_IP_PREFIX
+                ),
+            };
```

Comment:
When `ip_from` not defined, I got this in keyfile:

```
[ipv4]
address1=192.168.1.1/24
method=manual
route1=192.168.2.0/24,192.168.1.3,108
route1_options=table=200
routing-rule1=priority 30000 to 192.168.3.2 table 200

[ipv6]
addr-gen-mode=eui64
address1=2001:db8:1::1/64
method=manual
route1=2001:db8:a::/64,2001:db8:1::2,108
route1_options=table=200
routing-rule1=priority 30000 to 2001:db8:b::/64 table 200
```

Reply:
I noticed that the default `ip_from` is only set when the `ip_to` is not set. Let me fix that

Reply:
Done, now the default `ip_from` is only set when `ip_to` is also not set. Thanks!